src/libs/utils.ts

```diff
@@ -6,6 +6,7 @@ import * as qs from 'qs';
 import { div, span } from 'react-hyperscript-helpers';
 
 export {
+  append,
```

Comment:
Since the core-utils `append` function has a different interface than the `append` function that was defined in this module, I think we shouldn't re-export it. Consumers importing from this module may be expecting the old interface. Importing `append` from core-utils instead of libs/utils clarifies that it's a new function.

Reply:
I don't think I understand. By different interface, do you mean that this version is typed and the other version was not? Otherwise, I think both are curried functions that expects a single item and then return a function that expects a list.

Reply:
The `append` currently in libs/utils takes either 1 or 2 arguments. The `append` in core-utils takes only one. In order to re-export, we'd need to be sure that nothing calls `append` with two arguments.

Since the interface is different, I'd argue these are two different functions, despite sharing the same name. So while I think re-exporting makes sense for maintaining import paths when _moving_ a function to the core-utils package, I'd claim it's more appropriate to migrate consumers to the new interface by importing the new function from core-utils vs re-exporting and altering the behavior of libs/utils' `append`.

Before:
```js
append(3, [1,2])
// [1, 2, 3]
```

After:
```
append(3, [1,2])
// (list) => {
//    return [...list, value];
// }
```

Reply:
Oh ok, that makes sense. Thanks!

I didn't see any usages that incorporated two arguments, and I believe that the re-export would have broken the build if there were any.

I like your approach, and I think that is the right way to go. Just to consider all options though, would you consider it beneficial to have both signatures still available by rewriting append to something like this:

```
export function append<Type>(value: Type, list?: Type[]): ((list: Type[]) => Type[]) | Type[] {
  if(list){
    return [...list, value];
  }
  return curyAppend(value);
};

const curyAppend = <Type>(value: Type): ((list: Type[]) => Type[]) => {
  return (list: Type[]): Type[] => {
    return [...list, value];
  };
};
```

The benefits of this is that both signatures are still available, but the draw back is that the return type is a bit ambiguous.

Do you have thoughts here?

Reply:
> I didn't see any usages that incorporated two arguments, and I believe that the re-export would have broken the build if there were any.

There may not be any existing uses of that form. Though I think developers that have used the function before may assume the old signature. Importing from a different place and dropping the `Utils.` prefix serves as a (admittedly subtle) signal that the function has changed.

Also, I think TypeScript would catch any uses of the two argument version. But plain JavaScript will happily allow misuse.

If you did want to keep the same signature, I believe the easiest way to do that is to replace `_.curry` with `safeCurry`, which is `_.curry` with types.

Reply:
That is a good point. Thanks for pointing that out. It sounds as though implementing this will involve some refactoring (either changing imports or altering calls somehow). With that in mind, I am going to refactor the append function for readability and then adjust the callers as necessary. I have started the refactor here: https://github.com/DataBiosphere/terra-ui/pull/4597